tests/integrationv2/test_session_resumption.py

```diff
@@ -102,3 +100,133 @@ def test_session_resumption_s2n_client(managed_process, cipher, curve, protocol,
+    s2n_version = get_expected_s2n_version(protocol, OpenSSL)
+
+    for results in client.get_results():
+        assert results.exception is None
+        assert results.exit_code == 0
+        assert bytes("Protocol  : {}".format(openssl_version).encode('utf-8')) in results.stdout
+
+    # The server should indicate a session has been resumed
+    for results in server.get_results():
+        assert results.exception is None
+        assert results.exit_code == 0
+        assert b'Resumed session' in results.stdout
+        assert bytes("Actual protocol version: {}".format(s2n_version).encode('utf-8')) in results.stdout
+
+
+@pytest.mark.uncollect_if(func=invalid_test_parameters)
+@pytest.mark.parametrize("cipher", TLS13_CIPHERS, ids=get_parameter_name)
+@pytest.mark.parametrize("curve", ALL_TEST_CURVES, ids=get_parameter_name)
+@pytest.mark.parametrize("certificate", ALL_TEST_CERTS, ids=get_parameter_name)
+@pytest.mark.parametrize("protocol", [Protocols.TLS13], ids=get_parameter_name)
+@pytest.mark.parametrize("provider", [OpenSSL], ids=get_parameter_name)
+def test_tls13_session_resumption_s2n_client(managed_process, cipher, curve, protocol, provider, certificate):
+    port = str(next(available_ports))
+
+    random_bytes = data_bytes(64)
+
+    # The reconnect option for s2nc allows the client to reconnect automatically
+    # five times. In this test we expect one full connection and five resumption
+    # connections.
+    num_full_connections = 1
+    num_resumed_connections = 5
+
+    client_options = ProviderOptions(
+        mode=Provider.ClientMode,
+        host="localhost",
+        port=port,
+        cipher=cipher,
+        curve=curve,
+        insecure=True,
+        use_session_ticket=True,
+        data_to_send=random_bytes,
+        reconnect=True,
+        protocol=protocol)
+
+    server_options = copy.copy(client_options)
+    server_options.mode = Provider.ServerMode
+    server_options.key = certificate.key
+    server_options.cert = certificate.cert
+    server_options.reconnects_before_exit = num_resumed_connections + num_full_connections
+
+    server = managed_process(provider, server_options, timeout=5)
+    client = managed_process(S2N, client_options, timeout=5)
+
+    s2n_version = get_expected_s2n_version(protocol, OpenSSL)
+
+    # s2nc indicates the number of resumed connections in its output
+    for results in client.get_results():
+        assert results.exception is None
+        assert results.exit_code == 0
+        assert results.stdout.count(b'Resumed session') == num_resumed_connections
+        assert bytes("Actual protocol version: {}".format(s2n_version).encode('utf-8')) in results.stdout
+
+    # s_server only writes one certificate message in all of the connections
+    for results in server.get_results():
+        assert results.exception is None
+        assert results.exit_code == 0
+        assert bytes("6 server accepts that finished".encode('utf-8')) in results.stdout
```

Comment:
This six is actually (num_full_connections+num_resumed_connections), right? You might want to actually use the variables in the string.